src/components/SharedComponents/Button/button.styles.ts

```diff
@@ -268,14 +275,15 @@ const buttonStyles = css<ButtonStylesProps>`
     props.variant === "buttonStyledAsLink" &&
     props.size === "medium" &&
     css`
-      & svg[name="underline-1"] {
-        top: 27px;
-        width: 96%;
+      & ${ButtonStyledAsLinkFocusUnderline} {
+        display: block;
+        bottom: -1px;
+        left: ${props.$iconPosition === "leading" ? "4px" : ""};
+        width: ${props.icon ? "calc(100% - 4px)" : "100%"};
```

Comment:
Pass though `icon` so we know if it's got an icon or not and reduce the underline

Reply:
Good point